### PR TITLE
[Enhancement]support object storage that not compatible with s3

### PR DESCRIPTION
**Pull request by zombee0**
## Why I'm doing:
some object storage is not compatible with s3 sdk

## What I'm doing:
support access object storage without s3 sdk

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/fs/fs_util.h [hunk 1]
```diff
@@ -197,9 +198,10 @@ inline bool starts_with(std::string_view s, std::string_view prefix) {
 }
 
 inline bool is_s3_uri(std::string_view uri) {
-    return starts_with(uri, "oss://") || starts_with(uri, "s3n://") || starts_with(uri, "s3a://") ||
-           starts_with(uri, "s3://") || starts_with(uri, "cos://") || starts_with(uri, "cosn://") ||
-           starts_with(uri, "obs://") || starts_with(uri, "ks3://") || starts_with(uri, "tos://");
+    return starts_with(uri, "s3n://") || starts_with(uri, "s3a://") || starts_with(uri, "s3://") ||
+           (config::object_storage_use_aws_sdk_default &&
+           (starts_with(uri, "oss://") || starts_with(uri, "cos://") || starts_with(uri, "cosn://") ||
+           starts_with(uri, "obs://") || starts_with(uri, "ks3://") || starts_with(uri, "tos://")));
```
Comment (kevincai) [inline be/src/fs/fs_util.h:204]: what sdk/access client will be used for these oss/cos/... if the configure is turned on?

Comment (zombee0) [reply be/src/fs/fs_util.h:204]: will use hdfs protocol, and user need to put the vendor specified jar to be/lib/hdfs/ and add the auth information in core-site.xml

File: be/src/fs/fs_util.h [hunk 2]
```diff
@@ -197,9 +198,10 @@ inline bool starts_with(std::string_view s, std::string_view prefix) {
 }
 
 inline bool is_s3_uri(std::string_view uri) {
-    return starts_with(uri, "oss://") || starts_with(uri, "s3n://") || starts_with(uri, "s3a://") ||
-           starts_with(uri, "s3://") || starts_with(uri, "cos://") || starts_with(uri, "cosn://") ||
-           starts_with(uri, "obs://") || starts_with(uri, "ks3://") || starts_with(uri, "tos://");
+    return starts_with(uri, "s3n://") || starts_with(uri, "s3a://") || starts_with(uri, "s3://") ||
+           (config::object_storage_use_aws_sdk_default &&
```
Comment (sduzh) [inline be/src/fs/fs_util.h:202]: This configuration is a bit tricky and not very extensible, can we add some kind of mechanism that allows to specify the filesystem for a certain scheme through a configuration file? like this:
```
hdfs_scheme=obs,ks3
s3_scheme=oss,cos
```

Comment (sduzh) [reply be/src/fs/fs_util.h:202]: cc @kevincai

Comment (zombee0) [reply be/src/fs/fs_util.h:202]: in general case, these object storage should be compatible with s3, but in some case vendor may add some optimization that is not compatible with s3, in this case user can't use aws sdk, and use the hadoop jar that vendor supplied. what i mean is even same schema, some case should use aws sdk and some case not.

Comment (sduzh) [reply be/src/fs/fs_util.h:202]: > even same schema, some case should use aws sdk and some case not

It looks like that your current patch doesn't solve the problem?

Comment (kevincai) [reply be/src/fs/fs_util.h:202]: this is a global configure, it means that if the configure is turned on, all the related schema will be bypass s3 sdk and fallback to hdfs jar. what if they uses both tos and ks3 in the same cluster, and one is expecting to use s3 sdk and the other uses hdfs jar? can we do something to the broker job to help selecting the underlying fs?

**Comment by kevincai** [comment]
better add UT to cover the new function, with both the list is empty and the list is not empty. The CI code coverage result doesn't look like right to me.

**Comment by kevincai** [comment]
> #### [BE Incremental Coverage Report]
> ❌ **fail** : 8 / 12 (66.67%)
> 
> #### file detail
> path	covered_line	new_line	coverage	not_covered_line_detail
> 🔵	be/src/fs/fs.cpp	2	4	50.00%	[69, 94]
> 🔵	be/src/fs/fs_util.h	6	8	75.00%	[202, 203]

now it makes sense. lol

**Review by kevincai — APPROVED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Review by DorianZheng — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**